src/coguard_cli/image_check/config_file_finders/config_file_finder_tomcat.py

```diff
@@ -0,0 +1,242 @@
+        self.extract_web_xmls(
+            path_to_file_system,
+            os.path.join(path_to_file_system, web_xml_root_path[1:]),
+            temp_location_tuple
+        )
+        return temp_location_tuple
+
+    def check_for_config_files_filesystem_search(
+            self,
+            path_to_file_system: str
+    ) -> List[Tuple[Dict, str]]:
+        """
+        See the documentation of ConfigFileFinder for reference.
+        """
+        standard_names = ["server.xml", "web.xml", "context.xml"]
+        result_files = []
+        logging.debug("Attempting to find the standard names in the file system")
+        for (dir_path, _, file_names) in os.walk(path_to_file_system):
+            for standard_name in standard_names:
+                logging.debug("Checking if %s is in %s",
+                              standard_name,
+                              json.dumps(file_names))
+                if standard_name in file_names:
+                    result_files.append(os.path.join(dir_path, standard_name))
+        logging.debug("Found the following files with standard names in the file system: %s",
+                      result_files)
+        results = []
+        for result_file in result_files:
+            print(
+                f"{COLOR_CYAN}Found file "
+                f"{result_file.replace(path_to_file_system, '')}"
+                f"{COLOR_TERMINATION}"
+            )
+            results.append(self._create_temp_location_and_mainfest_entry(
+                path_to_file_system,
+                (os.path.basename(result_file), result_file)
+            ))
+        return results
+
+    def check_call_command_in_container(
+            self,
+            path_to_file_system: str,
+            docker_config: Dict
+    ) -> List[Tuple[Dict, str]]:
+        """
+        See the documentation of ConfigFileFinder for reference.
+        """
+        env_vars = docker_config.get(
+            "Config",
+            {}
+        ).get(
+            "Env",
+            []
+        )
+        catalina_home = reduce(
+            lambda acc, env_var: env_var.split("=")[1] \
+            if env_var.split("=")[0] == "CATALINA_HOME" else acc,
+            env_vars,
+            ""
+        )
+        results = []
+        if catalina_home:
+            config_files_in_standard_locations = [
+                os.path.join(catalina_home, 'conf/server.xml'),
+                os.path.join(catalina_home, 'conf/web.xml'),
+                os.path.join(catalina_home, 'conf/context.xml'),
+            ]
+            web_xml_root_path = os.path.join(catalina_home, 'webapps')
+            logging.debug(
+                "Attempting to find all the Tomcat configurations in CATALINA_HOME location"
+            )
+            temp_location_tuple = None
+            for config_file in config_files_in_standard_locations:
+                location_on_current_machine = os.path.join(path_to_file_system, config_file[1:])
```

Comment:
There is some code duplication going on. I suggest the following:

1. Create unit tests
2. put the duplication into its own function